FlowCrypt/Functionality/Services/EmailKeyManagerApi.swift

```diff
@@ -0,0 +1,55 @@
+//
+//  EmailKeyManagerApi.swift
+//  FlowCrypt
+//
+//  Created by Yevhen Kyivskyi on 17.07.2021.
+//  Copyright © 2021 FlowCrypt Limited. All rights reserved.
+//
+
+import Promises
+
+protocol EmailKeyManagerApiType {
+
+    func getPrivateKeysUrlString() -> String?
+    func getPrivateKeys() -> Promise<DecryptedPrivateKeysResponse>
+}
+
+class EmailKeyManagerApi: EmailKeyManagerApiType {
+
+    private let organisationalRulesService: OrganisationalRulesServiceType
+
+    init(organisationalRulesService: OrganisationalRulesServiceType = OrganisationalRulesService()) {
+        self.organisationalRulesService = organisationalRulesService
+    }
+
+    func getPrivateKeysUrlString() -> String? {
+        guard let keyManagerUrlString = organisationalRulesService.getSavedOrganisationalRulesForCurrentUser().keyManagerUrlString else {
+            return nil
+        }
+        return "\(keyManagerUrlString)v1/keys/private"
+    }
+
+    func getPrivateKeys() -> Promise<DecryptedPrivateKeysResponse> {
+        Promise<DecryptedPrivateKeysResponse> { [weak self] resolve, _ in
+            guard let self = self else { throw AppErr.nilSelf }
+            guard let urlString = self.getPrivateKeysUrlString() else {
+                resolve(.empty)
+                return
+            }
```

Comment:
If we are calling this method, then either we made the request out to the server, or we should resolve an error.

To choose whether to call or not should happen wherever this method is called.